build-system/tasks/storybook/amp-env/decorator.js

```diff
@@ -16,13 +16,27 @@
 
 import * as Preact from '../../../../src/preact';
 import {makeDecorator} from '@storybook/addons';
+import flush from 'styled-jsx/server';
 import render from 'preact-render-to-string/jsx';
 
 export default makeDecorator({
   name: 'withAmpDecorator',
   parameterName: 'amp',
   wrapper: (getStory, context) => {
     const contents = render(getStory(context));
+    const styleElements = flush();
+    const styles = render(
+      <style
+        amp-custom=""
+        dangerouslySetInnerHTML={{
+          __html: styleElements
+            .map((style) => style.props.dangerouslySetInnerHTML.__html)
+            .join('')
+            .replace(/\/\*# sourceMappingURL=.*\*\//g, '')
```

Comment:
Is it necessary to remove the source maps? Maybe they can be adjusted slightly?

Reply:
@jridgewell recently added a way to override this URL while fixing sourcemaps paths (#27665), but it may need you to build things differently.

Reply:
Since this output isn't going to be used anywhere outside storybook, I don't think it's necessary to involve gulp. Maybe we should just leave the source-maps actually?